src/Rules/FunctionDefinitionCheck.php

```diff
@@ -103,7 +124,7 @@ public function checkFunction(
 			$class = $param->type instanceof NullableType
 				? (string) $param->type->type
 				: (string) $param->type;
-			if ($class === '' || in_array($class, self::VALID_TYPEHINTS, true)) {
+			if ($class === '' || in_array($class, $this->getSupportedTypehints($function), true)) {
```

Comment:
The `in_array` search is unnecessary. You can change the `getSupportedTypehints` to `isTypeSupported($class, $function)` which would basically just return

~~~php
return isset($supportedTypes[$type]) && $supportedTypes[$type] <= PHP_VERSION_ID;
~~~